src/include/stir/listmode/LmToProjDataAbstract.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+//
+//
+/*!
+  \file 
+  \ingroup listmode
+
+  \brief Abstract base class for listmode to projection data conversion.
+ 
+  \author Richard Brown
+  
+*/
+/*
+    Copyright (C) 2020, University College of London
+    This file is part of STIR.
+
+    This file is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.
+
+    This file is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    See STIR/LICENSE.txt for details
+*/
+
+#include "stir/ParsingObject.h"
+
+START_NAMESPACE_STIR
+
+/*!
+  \ingroup listmode
+
+  \brief This class is the abstract base class fir binning listmode data to projection data,
+  i.e. (3d) sinograms.
+
+  It provides the basic machinery to go through a list mode data file,
+  and write projection data for each time frame. 
+*/
+
+class LmToProjDataAbstract : public ParsingObject
```

Comment:
should add `set_up()`

Reply:
`LmToProjData` doesn't have a `set_up` method, so can't do that unfortunately.

Reply:
sure you can. just add a default implementation `set_up()` which doesn't do anything :-)

Reply:
I've just realised that neither child class has a `set_up` method. So can add, but doesn't seem like it serves much purpose.

Reply:
it doesn't, but it prepares us for the future